node/impl/eth/transaction.go

```diff
@@ -114,101 +94,44 @@ func (e *ethTransaction) EthBlockNumber(ctx context.Context) (ethtypes.EthUint64
 	return ethtypes.EthUint64(parent.Height()), nil
 }
 
-func (e *ethTransaction) EthGetBlockTransactionCountByNumber(ctx context.Context, blkNum ethtypes.EthUint64) (ethtypes.EthUint64, error) {
-	ts, err := e.chainStore.GetTipsetByHeight(ctx, abi.ChainEpoch(blkNum), nil, false)
+func (e *ethTransaction) EthGetBlockTransactionCountByNumber(ctx context.Context, blkParam string) (ethtypes.EthUint64, error) {
+	ts, err := e.tipsetResolver.GetTipsetByBlockNumber(ctx, blkParam, true)
 	if err != nil {
-		return ethtypes.EthUint64(0), xerrors.Errorf("error loading tipset %s: %w", ts, err)
```

Comment:
the rule is now - don't wrap errors that come out of the `TipSetResolver` because they may be an `ErrNullRound` which won't travel through go-jsonrpc when wrapped, plus the extra wrapped text just makes the error weird and confusing

Reply:
This might deserve an in-line comment in code.